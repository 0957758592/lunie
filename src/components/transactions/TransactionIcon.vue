<template>
  <div class="tx__icon">
    <img :class="transactionGroup" src="~assets/images/cosmos-logo.png" />
  </div>
</template>

<script>
export default {
  name: `transaction-icon`,
  props: {
    transactionGroup: {
      type: String,
      required: true
    }
  }
}
</script>

<style>
.tx__icon {
  padding: 12px 0 12px 1rem;
}

.tx__icon img {
  max-height: 100%;
  max-width: 52px;
  border: 2px solid;
  border-radius: 50%;
  display: block;
  border-color: grey;
}

.tx__icon img.banking {
  border-color: #ed553b;
}

.tx__icon img.staking {
  border-color: #47ab6c;
}

.tx__icon img.governance {
  border-color: #15cfcc;
}

.tx__icon img.distribution {
  border-color: #f2b134;
}
</style>
