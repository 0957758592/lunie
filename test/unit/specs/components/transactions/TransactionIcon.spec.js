import { shallowMount } from "@vue/test-utils"
import TransactionIcon from "src/components/transactions/TransactionIcon"

describe(`TransactionIcon`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(TransactionIcon, {
      propsData: {
        group: "banking"
      }
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
