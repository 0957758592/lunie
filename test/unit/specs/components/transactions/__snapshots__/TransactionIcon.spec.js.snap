// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionIcon has the expected html structure 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="cosmic atom token"
    class="banking"
    src="~assets/images/cosmos-logo.png"
  />
</div>
`;
